### ISSUE TITLE
Migrate into llvm-hs-pure

### ISSUE THREAD

**Issue by sdiehl** [llvm-hs/llvm-irbuilder#7]
@cocreature What do you think about pulling this into the mainline `llvm-hs-pure` on the next major release? I think it maps closely to the one exported by the [C++ version](https://llvm.org/doxygen/classllvm_1_1IRBuilder.html) and makes sense to export the same functionality from Haskell bindings.

**Comment by cocreature** [comment]
Sounds good! Before that happens we should probably move the example to a test suite and I think it would be nice to move the module hierarchy to `LLVM.*` to be consistent with the rest of `llvm-hs-pure`.

**Comment by alpmestan** [comment]
Hello! Any plans for releasing this on hackage in the meantime?

**Comment by cocreature** [comment]
Hey @alpmestan, releasing a package to Hackage only to deprecate it shortly thereafter seems a bit weird. I’d rather accelerate merging this into `llvm-hs-pure` and make a release of that. The only technical step that should happen before that is changing the module hierarchy. However, I think it makes sense to two weeks or so for things to stabilize before we make a release.

Is there some specific reason for why you need a Hackage release? (Given that you use nix, depending on a package from Github should be straightforward :))

**Comment by sdiehl** [comment]
Indeed, we'll probably move it directly into `llvm-hs-pure` and cut a new release there. If you use Stack or Nix it should be easy to point at the Github for dev version. The new module will probably be ``LLVM.IRBuilder``.

**Comment by alpmestan** [comment]
Well, it's just that 2 of the 3 LLVM packages that I use, I get from github. There's no problem in that per se (a hackage release would however be nice: discoverability, readily available haddocks), but I'd definitely think that llvm-hs's companions, -pretty and -irbuilder, are useful enough as they are to deserve a release on hackage. But if you want to consolidate/merge/etc before going ahead that makes perfect sense, I can still use it without any problem. I was just inquiring :)

**Comment by sdiehl** [comment]
We'll move ``llvm-pretty``  to a unique package, I'm just finishing up the printer for a few of the quirkier corners of the language and then we'll put in Hackage.

**Comment by alpmestan** [comment]
Great, I'll keep an eye on it in the upcoming weeks/months :)

**Comment by sdiehl** [comment]
@alpmestan The pretty printer is available on Hackage now: https://hackage.haskell.org/package/llvm-hs-pretty

**Comment by alpmestan** [comment]
Lovely, thanks @sdiehl :)

**Comment by cocreature** [comment]
See https://github.com/llvm-hs/llvm-hs/pull/157 for an initial attempt at merging this into `llvm-hs-pure`.

**Comment by cocreature** [comment]
`llvm-irbuilder` is now merged into `llvm-hs-pure`. I’ll try to make a release sometime this week.

We should either delete this repository or at least make it very clear that this repository is no longer actively developed by mentioning in the README, disabling issues and maybe making a pull request template (apparently disabling pull requests is not possible). Given that this repository is relatively new, I think simply deleting it might be the best option but I don’t have a strong opinion on this. I’ll leave this issue open until we have chosen one of the options.

**Comment by ollef** [comment]
I need to update Sixten to not point to this repo. Would appreciate if it could stay for another week or so to make the transition smooth.

**Comment by sdiehl** [comment]
We'll mark it deprecated and then delete it in a week or so.

**Comment by cocreature** [comment]
There is now a Hackage release of `llvm-hs-pure` which includes the IRBuilder.
